SAMtags.tex

```diff
@@ -453,6 +453,165 @@ \subsubsection{Color space}
+processing software should consider that the tags may have local usage
+for other purposes.
+
+\begin{center}\small
+% This table is sorted alphabetically
+\begin{longtable}{ccp{12.5cm}}
+  \hline
+  {\bf Tag} & {\bf Type} & {\bf Description} \\
+  \hline
+  \endhead
+  {\tt Mm} & i & Base modifications / methylation  \\
+% {\tt MP} & i & Base modification qualities \\
+  {\tt Ml} & i & Base modification probabilities \\
+\end{longtable}
+\end{center}
+
+\subsection{Base modifications}
+
+Base modifications, including base methylation, are represented as a series of edits from the primary unmodified sequence as originally reported by the sequencing instrument.
+This potentially differs to the sequence stored in the main SAM {\sf SEQ} field if the latter has been reverse complemented, in which case SAM {\sf FLAG} 0x10 must be set.
+This means modification positions are also recorded against the original orientation (i.e. starting at the 5' end), and count the original base types.
+
+Each modified base listed also has a quality value associated with it.
+Given the unmodified base already has a phred likelihood, this base modification quality should be interpreted as the likelihood of this modification being correct given an assumption the original call is correct.
+
+\begin{description}
+\item[Mm:Z:\tagregex{([ACGTUN][-+][a-z]+(,[0-9]+)*;)*}]
+\hfill\\
+The first character is the unmodified ``fundamental'' base as reported
+by the sequencing instrument.
+It must be one of {\tt A}, {\tt C}, {\tt G}, {\tt T}, {\tt U} (if RNA) or {\tt N}.
+Note {\tt N} may be used to match any base rather than specifically an {\tt N} call by the sequencing instrument.
+This may be used in situations where the base modification is not a derivation of a standard base type.
+This is followed by either plus or minus indicating the strand the modification was observed on (relative to the original sequenced strand of {\sf SEQ} with plus meaning same orientation\footnote{Hence a tool that may reverse complement sequences does not need to understand how to manipulate the {\tt Mm} and {\tt Ml} tags.}), and one or more base modification symbols.
+This is then followed by a comma separated list of how many unmodified seq bases of the stated base type to skip, stored as a delta to the last and starting with 0 as the first (or next) base, starting from the uncomplemented 5' end of the {\sf SEQ} field.
+This number series is comparable to the numbers in an {\tt MD} tag,
+albeit counting specific base types only and potentially reverse-complemented.
+
+For example {\tt C+m,5,12,0;} tells us there are three 5-Methylcytosine bases in the original orientation {\sf SEQ}.
+The first 5 {\tt C} bases are unmodified and the 6th is modified, as are the 19th (with 12 between the 6th and 19th) and 20th.
+Similarly {\tt G-m,14;} indicates the 15th {\tt G} is a 5-Methylcytosine on the opposite strand (still counting from the 5' end).
```

Comment:
When noting a modification of the non-sequenced strand, do we still reference the base that was originally sequenced, or the rev-comp?

Reply:
I think that it would be good to have explicit examples for all 4 cases of nagativeStrand (in FLAG) and "ModifiedStrand" with bases so that it's clear which base is being modified...there are too many options for confusion.

Reply:
it would be good to state here that the abbreviation should _not_ be used.

Reply:
It was pointed out (maybe by you?) early on that the bases must be in the original orientation as reverse complementing a sequence eg via (re)alignment may not know about these tags.  So that bit hopefully is clear.   This question however is on the non-sequenced strand rather than the orientation (REVERSE sam flag).

It does need to be explicit, and yes it's confusing as I can't remember how it works myself and had to check!  The test/SAMtags/MM-orient.sam file has examples of all 4 combinations.  Do we want concrete examples in the spec too, or just wording to clarify this?

Note "non-sequenced strand" isn't perhaps the correct wording.  We cannot record modifications on the bottom strand unless it was sequenced.  The theory here is some technologies can do double stranded sequencing (eg the 1D^2 ONT which did top strand followed by bottom strand).  Single stranded sequencing is always going to report the same strand I'd guess, and reverse complement as appropriate.  I'll make some modifications to clarify this as I see I'm wrongly saying "original orientation" when I mean "top strand".

Reply:
> it would be good to state here that the abbreviation should _not_ be used.

I don't understand this comment, sorry.

Reply:
I am wondering though about the wisdom of my reporting using the fundamental (canonical) base on the top strand.  For example:

```
       m
AGCTTAGCGGCGATCGAG
||||||||||||||||||
TCGAATCGCCGCTAGCTC
               m
```

We'd report that as C+m and G-m as it's C and G on top strand.  It may be much clearer to say C-m for the bottom case as it's a C->m transition not G->m and then we're consistent.  I'm struggling to remember why I did it the way I did.  I fear it was for a good reason, but I cannot grasp what that would be!

Reply:
I'm not writing any software to output methylation on both strands yet, so happy with either resolution.